src/Symfony/Component/VarExporter/ProxyHelper.php

```diff
@@ -0,0 +1,337 @@
+        $parent = $class ? ' extends \\'.$class->name : '';
+        array_unshift($types, $class ? 'parent' : '');
+        $type = ltrim(implode('&\\', $types), '&');
+
+        if (!$class) {
+            $trait = new \ReflectionMethod(LazyProxyTrait::class, 'initializeLazyObject');
+            $body = \array_slice(file($trait->getFileName()), $trait->getStartLine() - 1, $trait->getEndLine() - $trait->getStartLine());
+            $body[0] = str_replace('): parent', '): '.$type, $body[0]);
+            $methods = ['initializeLazyObject' => implode('', $body).'    }'] + $methods;
+        }
+        $body = $methods ? "\n".implode("\n\n", $methods)."\n" : '';
+
+        if ($class) {
+            $propertyScopes = substr(self::exportPropertyScopes($class->name), 1, -6);
+            $body = <<<EOPHP
+
+                private const LAZY_OBJECT_PROPERTY_SCOPES = [
+                    'lazyObjectReal' => [self::class, 'lazyObjectReal', null],
+                    "\\0".self::class."\\0lazyObjectReal" => [self::class, 'lazyObjectReal', null],{$propertyScopes}
+                ];
+            {$body}
+            EOPHP;
+        }
+
+        return <<<EOPHP
+            {$parent} implements \\{$interfaces}
+            {
+                use \Symfony\Component\VarExporter\LazyProxyTrait;
+
+                private {$readonly}int \$lazyObjectId;
+                private {$readonly}{$type} \$lazyObjectReal;
+            {$body}}
+
+            // Help opcache.preload discover always-needed symbols
+            class_exists(\Symfony\Component\VarExporter\Internal\Hydrator::class);
+            class_exists(\Symfony\Component\VarExporter\Internal\LazyObjectRegistry::class);
+            class_exists(\Symfony\Component\VarExporter\Internal\LazyObjectState::class);
+
+            EOPHP;
+    }
+
+    public static function exportSignature(\ReflectionFunctionAbstract $function, bool $withParameterTypes = true): string
+    {
+        $parameters = [];
+        foreach ($function->getParameters() as $param) {
+            if (\in_array($default = rtrim(substr(explode('$'.$param->name.' = ', (string) $param, 2)[1] ?? '', 0, -2)), ['<default>', 'NULL'], true)) {
+                $default = 'null';
+            } elseif (str_contains($default, '\\') || str_contains($default, '::')) {
+                $default = implode('', array_map(fn ($part) => match ($part[0]) {
+                    '"' => $part, // for internal classes only
+                    "'" => str_replace("\0", '\'."\0".\'', $part),
+                    default => preg_replace('/(?<![a-zA-Z0-9_\x7f-\xff\\\\])[a-zA-Z0-9_\x7f-\xff]++(?:\\\\[a-zA-Z0-9_\x7f-\xff]++|::)++/', '\\\\\0', $part),
+                }, preg_split("/(\"(?:[^\"\\\\]*+(?:\\\\.)*+)*+\"|'(?:[^'\\\\]*+(?:\\\\.)*+)*+')/", $default, -1, \PREG_SPLIT_DELIM_CAPTURE | \PREG_SPLIT_NO_EMPTY)));
+            }
+            $parameters[] = ($param->getAttributes(\SensitiveParameter::class) ? '#[\SensitiveParameter] ' : '')
```

Comment:
Might be a dumb question but why not exporting all param attributes? Same for method-level attributes other than `ReturnTypeWillChange`?

Reply:
Could this be covered by tests btw?

Reply:
Configurable attributes would be much harder to re-apply on the generated code.

Reply:
@nicolas-grekas but if we read an attribute on `get_class($argument)`, we would still end up reading it on the generated proxy class

Reply:
For sure. I just have no idea why borrowing attributes from decorated to proxy would be the correct thing to do. What I know is that current proxies don't do this and it has never been an issue.

Reply:
(test case added)

Reply:
> What I know is that current proxies don't do this and it has never been an issue.

Fair. Future will tell then